cpp/include/Ice/ObjectAdapter.h

```diff
@@ -34,8 +34,9 @@ namespace Ice
         /**
          * Get the name of this object adapter.
          * @return This object adapter's name.
+         * @remark This function remains usable after the object adapter has been deactivated.
          */
-        [[nodiscard]] virtual std::string getName() const noexcept = 0;
+        [[nodiscard]] virtual std::string getName() const = 0;
```

Comment:
The implementation does not simply return _name, so no noexcept.

Reply:
Can't we use a constant for the unnamed OA name and always return a `cost string&`?

Reply:
Yes, will fix.